setup.py

```diff
@@ -331,7 +331,7 @@ def run(self):
 # All libraries that torch could depend on
 dep_libs = [
     'nccl', 'caffe2',
-    'libshm', 'libshm_windows', 'gloo', 'THD', 'c10d',
+    'libshm', 'libshm_windows', 'gloo',
```

Comment:
afaics, gloo and nccl have to come in at the same time as well.
Gloo and NCCL are dependencies of THD and c10d.

You're going have a harder time dealing with side-effects, than to simply move both gloo and nccl in this step.